src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchResultsLogger.kt

```diff
@@ -0,0 +1,84 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch
+
+import mu.KotlinLogging
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PagePrisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PrisonerMatches
+
+private val log = KotlinLogging.logger {}
+
+@Component
+class PrisonerSearchResultsLogger {
+
+  fun logResults(
+    prisonerMatches: PrisonerMatches,
+    prisonerSearchRequest: PrisonerSearchRequest,
+    caseload: String
+  ) {
+    log.info { "/match-prisoners returned ${prisonerMatches.matches.size} potential prisoner matches by ${prisonerMatches.matchedBy.value}" }
+    prisonerMatches.matches.forEach { it.prisoner.logData(prisonerSearchRequest, caseload, "/match-prisoners") }
+    prisonerMatches.matches.map { prisonerMatch -> prisonerMatch.prisoner }
+      .logMatchingSummary(prisonerSearchRequest, "/match-prisoners")
+  }
+
+  fun logResults(
+    pagePrisoner: PagePrisoner,
+    prisonerSearchRequest: PrisonerSearchRequest,
+    caseload: String
+  ) {
+    log.info { "/global-search found ${pagePrisoner.totalElements} potential prisoner matches" }
+    log.info { "/global-search page 1 of ${pagePrisoner.totalPages}" }
+    pagePrisoner.content?.forEach { it.logData(prisonerSearchRequest, caseload, "/global-search") }
+    pagePrisoner.content?.logMatchingSummary(prisonerSearchRequest, "/global-search")
+  }
+
+  private fun Prisoner.logData(prisonerSearchRequest: PrisonerSearchRequest, caseload: String, prefix: String) {
+    log.info { "$prefix prisoner: prisonNumber matched: ${if (prisonerSearchRequest.prisonNumber != null) prisonerSearchRequest.prisonNumber == prisonerNumber else "N/A"}" }
+    log.info { "$prefix prisoner: Status: $status" }
```

Comment:
I've added status, inoutStatus, legalStatus, lastMovementTypeCode and lastMovementReasonCode to the data we log about each prisoner that the search returns